qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/presenter.py

```diff
@@ -181,15 +181,19 @@ def action_copy_bin_to_table(self, table):
                         table_ws.setCell(j, 0, ws.getAxis(1).getValue(j))
                     else:
                         table_ws.setCell(j, 0, j)
-                table_ws.setCell(j, col_e, data_e[col])
-                table_ws.setCell(j, col_y, data_y[col])
+                # ragged workspaces could have data rows shorter than the number of columns
+                if col < len(data_e):
+                    table_ws.setCell(j, col_e, data_e[col])
```

Comment:
Rather than having this if statement each time, would it be nice to have a "set_cell_if_exists" function?

Reply:
Sorry to be pedantic, I think the function variable names make it confusing. Can I suggest:

```
def _set_cell_if_data_exists(
    target_table_ws,
    target_row_index,
    target_col_index,
    data_col,
    data_col_index,
):
    # ragged workspaces could have data rows shorter than the number of columns
    if data_col_index < len(data_col):
        target_table_ws.setCell(target_row_index, target_col_index, data[data_col_index])
```

Reply:
I was struggling on the names, so thanks for the suggestion 👍